File: test-fixtures/go/basic-functions.go
package main

import (
    "fmt"
    "errors"
    "context"
    "time"
)

// Basic function definitions
func greet(name string) string {
    return fmt.Sprintf("Hello, %s!", name)
}

func add(a, b int) int {
    return a + b
}

// Functions with multiple return values
func divide(a, b float64) (float64, error) {
    if b == 0 {
        return 0, errors.New("division by zero")
    }
    return a / b, nil
}

// Functions with named return values
func calculateStats(numbers []int) (sum, count int, average float64) {
    sum = 0
    count = len(numbers)
    for _, num := range numbers {
        sum += num
    }
    if count > 0 {
        average = float64(sum) / float64(count)
    }
    return
}

// Variadic functions
func sum(numbers ...int) int {
    total := 0
    for _, num := range numbers {
        total += num
    }
    return total
}

func printf(format string, args ...interface{}) {
    fmt.Printf(format, args...)
}

// Functions with pointers
func increment(x *int) {
    *x++
}

func swap(a, b *int) {
    *a, *b = *b, *a
}

// Functions with slices
func reverse(slice []int) []int {
    result := make([]int, len(slice))
    for i, v := range slice {
        result[len(slice)-1-i] = v
    }
    return result
}

func filterEven(numbers []int) []int {
    var result []int
    for _, num := range numbers {
        if num%2 == 0 {
            result = append(result, num)
        }
    }
    return result
}

// Functions with maps
func countWords(text string) map[string]int {
    words := make(map[string]int)
    // Simple word counting (would need proper tokenization)
    words["example"] = 1
    return words
}

// Functions with structs
type Person struct {
    Name string
    Age  int
}

func createPerson(name string, age int) Person {
    return Person{Name: name, Age: age}
}

func (p Person) greet() string {
    return fmt.Sprintf("Hello, I'm %s and I'm %d years old", p.Name, p.Age)
}

func (p *Person) birthday() {
    p.Age++
}

// Functions with interfaces
type Shape interface {
    Area() float64
    Perimeter() float64
}

type Rectangle struct {
    Width, Height float64
}

func (r Rectangle) Area() float64 {
    return r.Width * r.Height
}

func (r Rectangle) Perimeter() float64 {
    return 2 * (r.Width + r.Height)
}

func describeShape(s Shape) string {
    return fmt.Sprintf("Area: %.2f, Perimeter: %.2f", s.Area(), s.Perimeter())
}

// Generic functions (Go 1.18+)
func Max[T comparable](a, b T) T {
    if a > b {
        return a
    }
    return b
}

func Map[T, U any](slice []T, fn func(T) U) []U {
    result := make([]U, len(slice))
    for i, v := range slice {
        result[i] = fn(v)
    }
    return result
}

// Functions with channels
func producer(ch chan<- int) {
    for i := 0; i < 5; i++ {
        ch <- i
        time.Sleep(100 * time.Millisecond)
    }
    close(ch)
}

func consumer(ch <-chan int) {
    for value := range ch {
        fmt.Printf("Received: %d\n", value)
    }
}

// Functions with context
func processWithTimeout(ctx context.Context, data string) error {
    select {
    case <-time.After(2 * time.Second):
        fmt.Printf("Processed: %s\n", data)
        return nil
    case <-ctx.Done():
        return ctx.Err()
    }
}

// Functions with error handling
func readFile(filename string) ([]byte, error) {
    // Simulate file reading
    if filename == "" {
        return nil, errors.New("filename cannot be empty")
    }
    return []byte("file content"), nil
}

func processFile(filename string) error {
    data, err := readFile(filename)
    if err != nil {
        return fmt.Errorf("failed to read file %s: %w", filename, err)
    }
    fmt.Printf("File content: %s\n", string(data))
    return nil
}

// Functions with defer
func resourceManager() error {
    resource := "important resource"
    defer func() {
        fmt.Printf("Cleaning up: %s\n", resource)
    }()
    
    // Simulate some work
    fmt.Printf("Using: %s\n", resource)
    return nil
}

// Recursive functions
func factorial(n int) int {
    if n <= 1 {
        return 1
    }
    return n * factorial(n-1)
}

func fibonacci(n int) int {
    if n <= 1 {
        return n
    }
    return fibonacci(n-1) + fibonacci(n-2)
}

// Functions with panic and recover
func safeOperation() {
    defer func() {
        if r := recover(); r != nil {
            fmt.Printf("Recovered from panic: %v\n", r)
        }
    }()
    
    // This might panic
    panic("something went wrong")
}

// Higher-order functions
func applyOperation(a, b int, op func(int, int) int) int {
    return op(a, b)
}

func createMultiplier(factor int) func(int) int {
    return func(x int) int {
        return x * factor
    }
}

// Functions with type assertions
func processValue(v interface{}) {
    switch value := v.(type) {
    case int:
        fmt.Printf("Integer: %d\n", value)
    case string:
        fmt.Printf("String: %s\n", value)
    case bool:
        fmt.Printf("Boolean: %t\n", value)
    default:
        fmt.Printf("Unknown type: %T\n", value)
    }
}

// Main function
func main() {
    fmt.Println(greet("World"))
    fmt.Println(add(5, 3))
    
    result, err := divide(10, 2)
    if err != nil {
        fmt.Printf("Error: %v\n", err)
    } else {
        fmt.Printf("Result: %f\n", result)
    }
    
    fmt.Println(sum(1, 2, 3, 4, 5))
    
    p := createPerson("Alice", 30)
    fmt.Println(p.greet())
    p.birthday()
    fmt.Println(p.greet())
}

File: test-fixtures/java/BasicFunctions.java
package testfixtures;

import java.util.*;
import java.util.function.*;
import java.util.stream.Collectors;
import java.util.concurrent.CompletableFuture;
import java.io.IOException;

public class BasicFunctions {
    
    // Basic method definitions
    public static String greet(String name) {
        return "Hello, " + name + "!";
    }
    
    public static int add(int a, int b) {
        return a + b;
    }
    
    // Method overloading
    public static double add(double a, double b) {
        return a + b;
    }
    
    public static String add(String a, String b) {
        return a + b;
    }
    
    // Methods with different access modifiers
    public void publicMethod() {
        System.out.println("Public method");
    }
    
    private void privateMethod() {
        System.out.println("Private method");
    }
    
    protected void protectedMethod() {
        System.out.println("Protected method");
    }
    
    void packagePrivateMethod() {
        System.out.println("Package private method");
    }
    
    // Static methods
    public static void staticMethod() {
        System.out.println("Static method");
    }
    
    // Final methods
    public final void finalMethod() {
        System.out.println("Final method");
    }
    
    // Methods with generic types
    public static <T> T identity(T value) {
        return value;
    }
    
    public static <T> List<T> createList(T... elements) {
        return Arrays.asList(elements);
    }
    
    public static <T extends Comparable<T>> T max(T a, T b) {
        return a.compareTo(b) > 0 ? a : b;
    }
    
    // Methods with varargs
    public static int sum(int... numbers) {
        int total = 0;
        for (int num : numbers) {
            total += num;
        }
        return total;
    }
    
    // Methods with arrays
    public static int[] reverse(int[] array) {
        int[] result = new int[array.length];
        for (int i = 0; i < array.length; i++) {
            result[i] = array[array.length - 1 - i];
        }
        return result;
    }
    
    // Methods with collections
    public static List<Integer> filterEven(List<Integer> numbers) {
        return numbers.stream()
                     .filter(n -> n % 2 == 0)
                     .collect(Collectors.toList());
    }
    
    // Methods with functional interfaces
    public static <T, R> List<R> map(List<T> list, Function<T, R> mapper) {
        return list.stream()
                   .map(mapper)
                   .collect(Collectors.toList());
    }
    
    public static <T> List<T> filter(List<T> list, Predicate<T> predicate) {
        return list.stream()
                   .filter(predicate)
                   .collect(Collectors.toList());
    }
    
    // Methods with lambda expressions
    public static void processNumbers(List<Integer> numbers) {
        numbers.forEach(n -> System.out.println("Number: " + n));
        
        List<Integer> doubled = numbers.stream()
                                      .map(n -> n * 2)
                                      .collect(Collectors.toList());
        
        Optional<Integer> max = numbers.stream()
                                      .max(Integer::compareTo);
    }
    
    // Methods with exception handling
    public static String readFile(String filename) throws IOException {
        if (filename == null || filename.isEmpty()) {
            throw new IllegalArgumentException("Filename cannot be null or empty");
        }
        // Simulate file reading
        return "File content: " + filename;
    }
    
    public static String safeReadFile(String filename) {
        try {
            return readFile(filename);
        } catch (IOException e) {
            System.err.println("Error reading file: " + e.getMessage());
            return null;
        }
    }
    
    // Methods with multiple exceptions
    public static void validateData(String data) throws IllegalArgumentException, IllegalStateException {
        if (data == null) {
            throw new IllegalArgumentException("Data cannot be null");
        }
        if (data.length() < 3) {
            throw new IllegalStateException("Data too short");
        }
    }
    
    // Recursive methods
    public static int factorial(int n) {
        if (n <= 1) {
            return 1;
        }
        return n * factorial(n - 1);
    }
    
    public static int fibonacci(int n) {
        if (n <= 1) {
            return n;
        }
        return fibonacci(n - 1) + fibonacci(n - 2);
    }
    
    // Methods with annotations
    @Override
    public String toString() {
        return "BasicFunctions instance";
    }
    
    @Deprecated
    public static void oldMethod() {
        System.out.println("This method is deprecated");
    }
    
    @SuppressWarnings("unchecked")
    public static <T> T unsafeCast(Object obj) {
        return (T) obj;
    }
    
    // Async methods
    public static CompletableFuture<String> asyncGreet(String name) {
        return CompletableFuture.supplyAsync(() -> {
            try {
                Thread.sleep(1000); // Simulate delay
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
            return "Hello, " + name + "!";
        });
    }
    
    // Methods with builder pattern
    public static Builder builder() {
        return new Builder();
    }
    
    public static class Builder {
        private String name;
        private int age;
        
        public Builder name(String name) {
            this.name = name;
            return this;
        }
        
        public Builder age(int age) {
            this.age = age;
            return this;
        }
        
        public Person build() {
            return new Person(name, age);
        }
    }
    
    public static class Person {
        private final String name;
        private final int age;
        
        public Person(String name, int age) {
            this.name = name;
            this.age = age;
        }
        
        public String getName() {
            return name;
        }
        
        public int getAge() {
            return age;
        }
        
        @Override
        public String toString() {
            return String.format("Person{name='%s', age=%d}", name, age);
        }
    }
    
    // Methods with synchronized keyword
    public synchronized void synchronizedMethod() {
        System.out.println("Synchronized method");
    }
    
    public static synchronized void staticSynchronizedMethod() {
        System.out.println("Static synchronized method");
    }
    
    // Methods with native keyword (just declaration)
    public native void nativeMethod();
    
    // Methods with try-with-resources
    public static void processResource() {
        try (Scanner scanner = new Scanner(System.in)) {
            System.out.println("Processing input...");
            // Process input
        }
    }
    
    // Main method
    public static void main(String[] args) {
        System.out.println(greet("World"));
        System.out.println(add(5, 3));
        System.out.println(sum(1, 2, 3, 4, 5));
        
        List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5);
        List<Integer> evenNumbers = filterEven(numbers);
        System.out.println("Even numbers: " + evenNumbers);
        
        Person person = builder()
                .name("Alice")
                .age(30)
                .build();
        System.out.println(person);
        
        // Async example
        CompletableFuture<String> future = asyncGreet("Async World");
        future.thenAccept(System.out::println);
    }
}